core/src/main/java/org/bitcoinj/core/Wallet.java

```diff
@@ -1842,7 +1870,7 @@ private void receive(Transaction tx, StoredBlock block, BlockChain.NewBlockType
                 // Ignore the case where a tx appears on a side chain at the same time as the best chain (this is
                 // quite normal and expected).
                 Sha256Hash hash = tx.getHash();
-                if (!unspent.containsKey(hash) && !spent.containsKey(hash)) {
+                if (!unspent.containsKey(hash) && !spent.containsKey(hash) && !dead.containsKey(hash)) {
```

Comment:
Again - is this change right? If a tx appears in a side chain and is considered dead on the main chain, it should not go back into pending.